var/spack/repos/builtin/packages/chill/package.py

```diff
@@ -19,15 +19,16 @@ class Chill(AutotoolsPackage):
     version('0.3', sha256='574b622368a6bfaadbe9c1fa02fabefdc6c006069246f67d299f943b7e1d8aa3')
 
     depends_on('boost@1.66.0 cxxstd=11', type='build')
-    depends_on('rose@0.9.10.0 +cxx11', type='build')
+    depends_on('rose@0.9.13.0: +cxx11', type=('build', 'run'))
     depends_on('autoconf', type='build')
     depends_on('automake@1.14:',  type='build')
     depends_on('libtool', type='build')
     depends_on('m4', type='build')
-    depends_on('iegenlib', type='build')
-    depends_on('bison@3.4', type='build')
+    depends_on('iegenlib', type=('build', 'run'))
+    depends_on('bison@3.4.2:', type='build')
     depends_on('flex', type='build')
-    depends_on('python')
+    # Does not currrently work with Python3
+    depends_on('python@2.7.18:2.7.99')
```

Comment:
```suggestion
    depends_on('python@2.7:2.8')
```
I assume any version of Python 2.7 should work

Reply:
I am seeing both version span types within Spack. Are there guidelines for when to use one over the other?

Reply:
For the start of the version range:

What you're currently using only allows it to build with Python 2.7.18+, but 2.7.0-2.7.17 should work fine.

For the end of the version range:

This is where it gets controversial. Spack uses inclusive version ranges, so you usually want to use something like 2.7.999 or 2.7.develop. But since 2.8 doesn't exist, and never will, it's fine to use 2.8 in this case.

Reply:
Thank you for the explanation. Recommended changes submitted.